pkgs/os-specific/linux/dracut/default.nix

```diff
@@ -0,0 +1,128 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, gitUpdater
+, makeBinaryWrapper
+, pkg-config
+, asciidoc
+, libxslt
+, docbook_xsl
+, bash
+, kmod
+, binutils
+, busybox
+, bzip2
+, coreutils
+, cpio
+, findutils
+, glibc
+, gnugrep
+, gnused
+, gnutar
+, gzip
+, kbd
+, lvm2
```

Comment:
Technically you don't just remove optional dependencies. It makes sense for nix to have multiple entrypoints, so you can have a base package with just the minimum, then have flags that enable various features, then provide entrypoints that enable those flags as alternative entrypoints.

See things like vim, which as a `vimHugeX` variant.

Reply:
This could be done in combination with a nixos module where we know if we are booting from lvm2 or not. But generally we usually include all optional dependencies to reduce variations in dependencies.

NixOS does not support anything other than systemd, so unless you are using dracut to generate some other initramfs, it is always required.

Reply:
Yeah I thought about it already and decided to remove most of them to reduce closure size of the package, since I was going to change the DRACUT_PATH var to use `--suffix` and downstream packages/modules could easily wrap it with other requirements. I just haven't gotten around to it so assistance would be helpful :)

I mostly had them there to test if initrd generation even worked at all from the package (which surprisingly it sorta did, but still would need tons of conf to have it use stuff from the nix store)